feat(extensions): introduce Claude chat extension

dfseifert opened the pull request:
First of all, thanks a lot for this project!
I have been daily driving vicinae for a while now and just love it!

So I wanted to give back with creating a little extension I find quite useful.

This PR adds a new Claude chat extension for Vicinae. 

It enables AI-powered assistance through Anthropic's API, including features like persistent chat history, multiple model support, and real-time streaming.

> **Note:** This is an early example of a chat integration built with Vicinae extensions. It demonstrates how to build conversational AI experiences using Vicinae's native architecture.
> I am happy for any feedback!

<img width="1592" height="987" alt="image" src="https://github.com/user-attachments/assets/29019944-3b6d-4e49-805d-ada8c8ea0a20" />


<img width="1592" height="987" alt="image" src="https://github.com/user-attachments/assets/0792a838-7a94-456a-b53e-9d49c64624cc" />

dfseifert committed:
feat(extensions): introduce Claude chat extension

Added a new Claude chat extension for Vicinae. Enables AI-powered assistance through Anthropic's API, including features like persistent chat history, multiple model support, and real-time streaming.

dfseifert committed:
refactor(extensions): rename extension icon file

dfseifert committed:
refactor(chat): enhance error handling, message streaming, and chat storage updates

dfseifert commented:
Hi @aurelleb 

thanks for the feedback.

I've applied a patch that fixes the following:
The default view for the new chat command is now the chat history so that when navigating back, the chat history is shown.
Also when an Error during response collection occurs, it creates a new conversation item that displays the full error message. This could then be debugged. (Or is there a better way to do this?)
Please let me know, happy for feedback!

Kind Regards
Daniel

dfseifert committed:
Merge branch 'vicinaehq:main' into main

aurelleb commented:
> The default view for the new chat command is now the chat history so that when navigating back, the chat history is shown

I think this is weird behavior because it pushes a view on the stack without the user doing anything which makes the back navigation feel misleading. The average user will be confused about this.
You probably should have a command to start a new chat and a separate one that allows you to browse the existing threads.


https://github.com/user-attachments/assets/269e75fe-4bf6-4885-96bc-0d362fc6457d

aurelleb reviewed (changes requested):
see comment above

dfseifert committed:
refactor(chat): simplify new chat command and improve chat list actions

dfseifert committed:
feat(chat): add AI-generated titles for conversations

Integrated automatic title generation using Claude based on the first exchange. Simplified title handling by removing outdated logic and improving service helpers. Updated documentation to reflect this feature.

dfseifert committed:
refactor(chat): simplify action properties for consistency

dfseifert committed:
refactor(chat): modularize services and utilities, extract chat history hook

Streamlined code by refactoring chat storage, simplifying services into `ChatStorage` object with unified methods. Introduced `useChatHistory` hook for chat list management, improving modularity and reducing redundancy. Enhanced message utilities with reusable helpers.

dfseifert committed:
docs(readme): update shortcuts for deleting chats and improve clarity
refactor(chat-list): revise keyboard shortcut syntax for new chat action

dfseifert committed:
refactor(chat): improve hooks, storage reactivity, and action modularity

Refactored `useChat` and `useChatHistory` hooks for better reactivity and modularity by integrating `ChatStorage` subscriptions. Simplified chat storage methods and enhanced reusability. Updated action properties and shortcuts for consistency across components.

dfseifert committed:
refactor(chat): standardize formatting, improve reactivity, and cleanup hooks

Aligned code formatting across files for improved consistency. Enhanced storage reactivity by refining subscription logic in `useChat` and `useChatHistory`. Cleaned up hooks, modularized utility functions, and streamlined services for better maintainability and readability.

dfseifert committed:
refactor(chat): standardize code formatting and refine reactivity in hooks

Ensured consistent formatting across `useChat`, `useChatHistory`, and related services. Enhanced reactivity in storage subscriptions and improved maintainability by aligning indentation and structure. Updated README for clarity on streaming settings.

dfseifert committed:
refactor(chat-list): extract `makeChatKey` utility for better readability and reuse

dfseifert committed:
Merge branch 'vicinaehq:main' into main

dfseifert committed:
Merge branch 'vicinaehq:main' into main
